server/server.go

```diff
@@ -269,9 +263,27 @@ func (s *server) Boot() {
 					wrappedEndpoint := s.newEndpointWrapper(e)
 					wrappedEncoder := s.newEncoderWrapper(e, responseWriter)
 
+					// Combine all options this server defines. Since the interface of the
+					// go-kit server changed to not accept a context anymore we have to
+					// work around the context injection by injecting our context via the
+					// very first request function.
+					//
+					// NOTE this is rather an ugly hack and should be revisited. It would
+					// probably make sense to start decoupling from the go-kit code since
+					// there haven't been any benefits from its implementation, but only
+					// from its design ideas. Also note that some of the design ideas
+					// dictated by go-kit do not align with our own ideas and often stood
+					// in our way of making things work how they should be.
+					options := []kithttp.ServerOption{
+						kithttp.ServerBefore(func(context.Context, *http.Request) context.Context {
+							return ctx
+						}),
+						kithttp.ServerBefore(s.requestFuncs...),
+						kithttp.ServerErrorEncoder(s.newErrorEncoderWrapper()),
+					}
```

Comment:
This is rather odd. It works, but as mentioned in the go comment above I think we should improve this over time and move away from the hard bindings dictated by go-kit. We can keep some design patterns and ideas that align with us and work for us, but we should define our own types and interfaces and implementing the middleware handling is not more than a loop as well.